README.md

```diff
@@ -1,52 +1,226 @@
+    event.response.status(200).json({ totalCount: todos.rowCount })
+  } else {
+    event.response.status(400).json({ 'message': 'Bad request. We only have todos' })
+  }
+})
+
+app.start()
+```
+
+### Table of Contents
+
+[Configuration Options](#configuration)
+
+[Events](#events)
+
+[Actions](#actions)
+
+[Utilitiey Functions](#utils)
+
+### <a name="configuration"></a> Configuration options
+
+```js
+const app = new Reshuffle()
+const wix = new WixConnector(app, {
+  secret: process.env.RESHUFFLE_WIX_SECRET,
+  webhookPath: process.env.RESHUFFLE_WIX_WEBHOOK,
+})
+```
+Both `secret` and `webhookPath` are optional.
+
+The secret is a string containing a shared secret as described in [Wix Authentication](https://www.wix.com/corvid/reference/spis/external-database-collections/external-database-collections/authentication)
+
+When configuring the external database collection on Wix, you can define a `settings` object that is sent from Wix to the connector with every request.
+
+To use (the optional) secret, you should configure the settings object to include:
+```json
+{
+  "secret": ".....your secret....."
+}
+```
+Wix then includes it in the incoming request like so:
+```json
+{
+    "requestContext": {
+        "settings": {
+            "secret": ".....your secret....."
+        },
+        "instanceId": "...",
+        "installationId": "...",
+        "memberId": "...",
+        "role": "..."
+  }
+}
+``` 
+If you do not define a secret in the connector configuration, then it will not expect one.
+
+You can use the `webhookPath` to configure the url that Wix hits when it makes its calls to
+your external database collection. The value of `webhookPath` will be appended to your runtime's
+base url.
+Left unprovided, `webhookPath` defaults to `/webhooks/wix`, so if your runtime runs at `https://example.com/` then
+your full webhook path is `https://example.com/webhooks/wix`. This is the path you'll need
+to register with Wix when defining a new external collection.
+
+### <a name="events"></a> Events
+
+When Wix makes a call to the external collection, the connector captures these calls. 
+It then triggers the corresponding event.
+
+To listen to events coming from Wix, you'll need to capture them with the connector's `on`
+function, providing a `WixEventConfiguration` to it.
+
+```typescript
+interface WixConnectorEventOptions {
+  action: WixAction // See below
+}
+```
+The connector triggers events of the following type:
 
-#### Configuration Options:
 ```typescript
-interface _CONNECTOR_NAME_ConnectorConfigOptions {
-  foo: string // foo description
-  bar?: number // bar description
+interface WixEvent {
+  requestContext: WixRequestContext
+  collectionName?: string
+  filter?: string
+  sort?: any
+  skip?: number
+  limit?: number
+  itemId?: string
+  item?: any
+  body?: any
+  action: WixAction
+  request: Request // The http request from Wix
+  response: Response // The http response object 
+}
+
+// Where...
+type WixAction =
```

Comment:
I'll move this block in just below `interface WixConnectorEventOptions {
  action: WixAction // See below
}`

Reply:
done